tool/tctl/common/plugin/netiq.go

```diff
@@ -0,0 +1,503 @@
+	oAuthClientSecret     string
+	insecureSkipVerify    bool
+}
+
+func (p *PluginsCommand) netIQSetupGuide(ctx context.Context) (netIQSettings, error) {
+	settings := netIQSettings{}
+	var err error
+
+	settings.ospURL, err = promptForURL(ctx, os.Stdout, netIQStep1Template, "Please enter the IDM OSP address", p.install.netIQ.insecureSkipVerify,
+		func(ctx context.Context, s string, b bool) error {
+			_, err := checkNetIQOSPAddress(ctx, s, b)
+			return trace.Wrap(err)
+		})
+	if err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+
+	settings.apiURL, err = promptForURL(ctx, os.Stdout, netIQStep2Template, "Please enter the IDM API address", p.install.netIQ.insecureSkipVerify, checkUnauthenticatedNetIQAPIAddress)
+	if err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+
+	if err := promptForContinue(os.Stdout, netIQStep3Template); err != nil {
+		return netIQSettings{}, err
+	}
+
+	settings.oAuthClientID, err = promptForInput(ctx, os.Stdout, netIQStep4Template, "Enter the OAuth ClientID")
+	if err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+
+	settings.oAuthClientSecret, err = promptForPassword(ctx, os.Stdout, "Enter the OAuth Client Secret")
+	if err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+
+	settings.identityVaultUser, err = promptForInput(ctx, os.Stdout, netIQStep5Template, "Enter the IDM User")
+	if err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+
+	settings.identityVaultPassword, err = promptForPassword(ctx, os.Stdout, "Enter the IDM User Password")
+	if err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+
+	settings.insecureSkipVerify = p.install.netIQ.insecureSkipVerify
+
+	// Validate the NetIQ settings.
+	if err := checkAuthenticatedNetIQAPIAddress(ctx, settings); err != nil {
+		return netIQSettings{}, trace.Wrap(err)
+	}
+	return settings, nil
+}
+
+func (p *PluginsCommand) InstallNetIQ(ctx context.Context, args installPluginArgs) error {
+	settings, err := p.netIQSetupGuide(ctx)
+	if err != nil {
+		if errors.Is(err, errCancel) {
+			return nil
+		}
+		return trace.Wrap(err)
+	}
+
+	plugin, err := createNetIQPlugin(&settings)
+	if err != nil {
+		return trace.Wrap(err, "failed to create NetIQ plugin")
+	}
+
+	creds, err := getNetIQPluginCredentials(settings)
+	if err != nil {
+		return trace.Wrap(err, "failed to get NetIQ plugin credentials")
+	}
+
+	createPluginRequest := &pluginspb.CreatePluginRequest{
+		Plugin:                plugin,
+		StaticCredentialsList: creds,
+		CredentialLabels: map[string]string{
+			netIQOrgURLLabel: settings.apiURL,
+		},
+	}
+
+	if _, err = args.plugins.CreatePlugin(ctx, createPluginRequest); err != nil {
+		return trace.Wrap(err, "failed to create NetIQ plugin")
+	}
+
+	fmt.Println("NetIQ plugin has been successfully installed.")
+
+	return nil
+}
+
+func promptForURL(ctx context.Context, out io.Writer, template, prompt string, insecureSkipVerify bool, checkFunc func(context.Context, string, bool) error) (string, error) {
```

Comment:
Do we have any shared code in tctl for prompting for input that we can reuse?

If not, should we create a prompt package?

If I was looking to create some code that needed to prompt the user I don't know that I would go looking in a file called netiq.go for it..